vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -192,6 +199,21 @@ func resourceVSphereVirtualMachine() *schema.Resource {
 	}
 }
 
+// VAppSchema represents the schema for the vApp sub-resource.
+//
+// This sub-resource allows the customization of vApp properties
+// on cloned VMs.
+func VAppSubresourceSchema() map[string]*schema.Schema {
```

Comment:
I think we could probably just put this directly into `schemaVirtualMachineConfigSpec` as it's so small, along with the `vapp` key. I'm almost tempted to also just say we flatten it and name it `vapp_properties`, but we might need to add more keys at a later time.